### ISSUE TITLE
Memory leaks

### ISSUE THREAD

**Issue by melancthon** [zwb-ict/node-zstd#3]
The following code generates memory leaks and eventually crashes.

I've tried it with node v11.10.0 and v10.15.1LTS on both Windows 7 SP1 and CentOS 7.

Any ideas?

```
const zstd = require("node-zstd")

for(var i =0; i< 1000000; i++){
	var s = "0123456789"	
	var c = zstd.compressSync(Buffer.from(s))
	var d = zstd.decompressSync(c)

	if(i%100==0){
		var res = process.memoryUsage()
		console.log("mem:", res.external.toLocaleString(), res.heapTotal.toLocaleString(), res.heapUsed.toLocaleString(), res.rss.toLocaleString())
	}
}
```

**Comment by gesh** [comment]
Hey @melancthon,
Did you manage to fix/prevent the memory leaks?

**Comment by melancthon** [comment]
> 
> 
> Hey @melancthon,
> Did you manage to fix/prevent the memory leaks?

No, I haven't looked into it more deeply.  However I'm still interested in seeing if there is a solution out there.

**Comment by julrich1** [comment]
I know this is an old issue but the async methods don't appear to have the same memory leak issue if you want to use those as a workaround. I'm guessing the root cause is somewhere in the bindings.
